torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1155,6 +1182,11 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
                        SkipInfo('TestCommon', 'test_variant_consistency_jit',
                                 device_type='cuda', dtypes=[torch.float16]),
                    )),
+    OpInfo('diff',
+           op=torch.diff,
+           dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           sample_inputs_func=sample_inputs_diff,
+           test_inplace_grad=False),
```

Comment:
Not for this PR, but the test for inplace grads should be skipped automatically if the op doesn't have an inplace variant.